docs/UPGRADING.md

```diff
@@ -4,6 +4,101 @@ Here you can find a list of migration guides to handle breaking changes between
 
 ## Unreleased
 
+### Change public library interface
+
+#### `builder` package
+
+This part refers to the `builder` package declared in `arduino/builder/`.
+
+`GenBuildPath()` function has been moved to package `sketch` in `arduino/sketch/sketch.go`. The signature is unchanged.
```

Comment:
```suggestion
#### `github.com/arduino/arduino-cli/arduino/builder` package

`GenBuildPath()` function has been moved to the `github.com/arduino/arduino-cli/arduino/sketch` package. The signature
is unchanged.
```
It seems like this way of referring to the packages is more clear to the type of reader who will be interested in the information. What do you think?

Reply:
Good idea, I didn't think about it.

Reply:
Since you are in favor of this change, I added suggestions below for the equivalent changes to the rest of the `UPGRADING.md` content.